libraries/chain/include/eosio/chain/platform_timer.hpp

```diff
@@ -26,17 +26,21 @@ struct platform_timer {
       callback still execute if the timer expires and stop() is called nearly simultaneously.
       However, set_expiration_callback() is synchronized with the callback.
    */
-   void set_expiration_callback(void(*func)(void*), void* user) {
+   void set_expiration_callback(void(*func)(void*), void* user, bool appending = false) {
       bool expect_false = false;
       while(!atomic_compare_exchange_strong(&_callback_variables_busy, &expect_false, true))
          expect_false = false;
       auto reset_busy = fc::make_scoped_exit([this]() {
          _callback_variables_busy.store(false, std::memory_order_release);
       });
-      EOS_ASSERT(!(func && _expiration_callback), misc_exception, "Setting a platform_timer callback when one already exists");
+      EOS_ASSERT(!(!appending && func && !_expiration_callbacks.empty()), misc_exception, "Setting a platform_timer callback when one already exists");
 
-      _expiration_callback = func;
-      _expiration_callback_data = user;
+      if (!func && !_expiration_callbacks.empty()) {
+         _expiration_callbacks.pop_back();
+         return;
+      }
+
+      _expiration_callbacks.push_back({func, user});
```

Comment:
It does seem kind of unfortunate we need to maintain a stack of callbacks when it seems like there will only effectively be 2 unique ones registered. We could remove the stack with something like a `swap_expiration_callback()` and place the onus on restoring the value on whoever did the swap (much like we have a requirement whoever registers a callback today clears it). Some other approaches I can think of maybe go against  JITs current design (for example instead of wiring up the callback to a timed_run we somehow wire it up directly to the backend allocator). I guess we can keep this as is for now but it'd be nice to find something better.